dace/properties.py

```diff
@@ -415,7 +415,8 @@ def initialize_properties(obj, *args, **kwargs):
         # Assert that there are no fields in the object not captured by
         # properties, unless they are prefixed with "_"
         for name, prop in obj.__dict__.items():
-            if name not in properties and not name.startswith("_"):
+            if (name not in properties and not name.startswith("_")
```

Comment:
It is still a bit ugly, can we completely avoid operation on strings here?

Reply:
This is how properties work (@definelicht). We say that anything that starts with `self._` is a private field and thus should not be saved with the serialization etc. There is no way of avoiding it without breaking all the code we currently have.

Reply:
Changing this would be a huge PR on its own, so if we want to consider this, it should be a PR of its own.

Reply:
OK then I'll resolve this for now and queue up another PR for this. We should consider existing projects like Python's built-in dataclasses and pydantic.